internal/paascharm/go.tmpl

```diff
@@ -0,0 +1,127 @@
+// Code generated by paascharmgen; DO NOT EDIT.
+package {{ .PackageName }}
+
+type CharmConfig struct {
+     ConfigOptions ConfigOptions
+     ProxyConfig   ProxyConfig
+     {{- if .Integrations }}
+         Integrations Integrations
+     {{- end }}
+}
+
+type ConfigOptions struct {
+     BaseURL string `env:"APP_BASE_URL"`
+     Port int `env:"APP_PORT"`
+     MetricsPort *int `env:"APP_METRICS_PORT"`
+     MetricsPath *string `env:"APP_METRICS_PATH"`
+     SecretKey *string `env:"APP_SECRET_KEY"`
+     {{- if .Options }}
+         UserConfigOptions
+     {{- end }}
+}
+
+{{ if .Options }}
+    type UserConfigOptions struct {
+    {{- range .Options }}
+        {{ .GoVarName }} {{ .GoVarType }} `env:"{{ .EnvVarName }}"`
+    {{- end }}
+    }
+{{- end }}
+
+type ProxyConfig struct {
+     HTTPProxy *string `env:"HTTP_PROXY"`
+     HTTPSProxy *string `env:"HTTPS_PROXY"`
+     NoProxy []string `env:"NO_PROXY"`
+}
+
+{{ if .Integrations }}
+   type Integrations struct {
+   {{- range .Integrations }}
+       {{ .GoName }} {{ .GoName }}Integration {{ if .IsDatabase }}`envPrefix:"{{ .DatabasePrefix }}"`{{ end }}
+   {{- end }}
+   }
+{{ end }}
+
+
+{{ if .HasDatabaseIntegrations }}
+
+   {{- range .Integrations }}
+       {{ if .IsDatabase }}
+       type {{ .GoName }}Integration struct {
+            DatabaseIntegration
+       }
+       {{ end }}
+    {{ end }}
```

Comment:
Can't this be done within the previous `.Integrations` range to avoid iterating through it twice ?

Reply:
The previous `range` is inside a struct and this is outside of another one, so I think that is not possible.